### ISSUE TITLE
PPOTrainer Appears to incorrectly handle `pad_token_id`

### ISSUE THREAD

**Issue by lapp0** [huggingface/trl#1482]
https://github.com/huggingface/trl/blob/main/trl/trainer/ppo_trainer.py#L534-L544

https://github.com/huggingface/trl/blob/main/trl/trainer/ppo_trainer.py#L558

This code creates an "all-ones" `attention_mask` however we need the pad token to be zeros and consider `tokenizer.pad_token_id`, not `tokenizer.eos_token_id`

I came across this issue because I discovered that my output quality was low, and when I removed all pad tokens the output was high quality.

**Comment by lapp0** [comment]
Disregard, the solution is to ensure you don't apply padding when passing the token ids to `ppo_trainer.generate`
